crates/astria-sequencer/src/bridge/bridge_sudo_change_action.rs

```diff
@@ -0,0 +1,181 @@
+use anyhow::{
+    ensure,
+    Context as _,
+    Result,
+};
+use astria_core::{
+    primitive::v1::Address,
+    protocol::transaction::v1alpha1::action::BridgeSudoChangeAction,
+};
+use tracing::instrument;
+
+use crate::{
+    accounts::state_ext::StateWriteExt as _,
+    bridge::state_ext::{
+        StateReadExt as _,
+        StateWriteExt as _,
+    },
+    state_ext::{
+        StateReadExt,
+        StateWriteExt,
+    },
+    transaction::action_handler::ActionHandler,
+};
+
+#[async_trait::async_trait]
+impl ActionHandler for BridgeSudoChangeAction {
+    async fn check_stateful<S: StateReadExt + 'static>(
+        &self,
+        state: &S,
+        from: Address,
+    ) -> Result<()> {
+        ensure!(
+            state
+                .is_allowed_fee_asset(self.fee_asset_id)
+                .await
+                .context("failed to check allowed fee assets in state")?,
+            "invalid fee asset",
+        );
+
+        // check that the sender of this tx is the authorized sudo address for the bridge account
+        let Some(sudo_address) = state
+            .get_bridge_account_sudo_address(&self.bridge_address)
+            .await
+            .context("failed to get bridge account sudo address")?
+        else {
+            // TODO: if the sudo address is unset, should we still allow this action
+            // if the sender if the bridge address itself?
```

Comment:
Probably otherwise you can never update bridge accounts created previous to this change?

Reply:
i left this as-is because i wasn't sure if we were going to hard-fork or regenesis. if we regenesis, better to leave this